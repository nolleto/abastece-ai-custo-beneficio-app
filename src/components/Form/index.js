import React, { useState } from 'react';
import Field from '../Field';
import FuelPrice from '../FuelPrice';
import './style.css';

const Form = () => {
  const [fuelPrice, setFuelPrice] = useState()
  const [requestPrice, setRequestPrice] = useState()
  const [realPricePaid, setRealPricePaid] = useState()

  return (
    <div className='form'>
      <Field
        label='Valor do combustível'
        description='Valor do combustível divulgado no posto sem km de vantagem'
        value={fuelPrice}
        setter={setFuelPrice}
      />

      <Field
        label='Valor total a pagar na bomba'
        description='Valor solicitado no app Abastece Ai'
        value={requestPrice}
        setter={setRequestPrice}
      />

      <Field
        label='Valor pago com desconto do km de vantagem'
        description='Valor pago na app Abastece Ai'
        value={realPricePaid}
        setter={setRealPricePaid}
      />

      <FuelPrice
        fuelPrice={fuelPrice}
        requestPrice={requestPrice}
        realPricePaid={realPricePaid}
      />
    </div>
  )
}

export default Form;
