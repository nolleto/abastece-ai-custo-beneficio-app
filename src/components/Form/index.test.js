import React from 'react';
import Form from './index';
import { shallow } from 'enzyme';

describe('Form component', () => {
  it('matches snapshot', () => {
    const snapshot = shallow(<Form />);

    expect(snapshot).toMatchSnapshot();
  });
});
