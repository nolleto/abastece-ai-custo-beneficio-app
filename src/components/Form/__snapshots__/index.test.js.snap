// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component matches snapshot 1`] = `
<div
  className="form"
>
  <Field
    description="Valor do combustível divulgado no posto sem km de vantagem"
    label="Valor do combustível"
    setter={[Function]}
  />
  <Field
    description="Valor solicitado no app Abastece Ai"
    label="Valor total a pagar na bomba"
    setter={[Function]}
  />
  <Field
    description="Valor pago na app Abastece Ai"
    label="Valor pago com desconto do km de vantagem"
    setter={[Function]}
  />
  <FuelPrice />
</div>
`;
